addons/point_of_sale/models/pos_order.py

```diff
@@ -170,6 +170,41 @@ def _prepare_invoice(self):
             'user_id': self.env.uid,
         }
 
+    @api.model
+    def _get_account_move_line_group_data_type_key(self, data_type, values):
+        """
+        Return a tuple which will be used as a key for grouping account
+        move lines in _create_account_move_line method.
+        :param data_type: 'product', 'tax', ....
+        :param values:
+        :return: tuple() representing the data_type key
+        """
+        build_key_method_name = '_get_account_move_line_%s_key' % data_type
```

Comment:
I changed and I just add one method. I understand my changes could reduce the readability and they were not needed at the moment.